src/sentry/tasks/data_export.py

```diff
@@ -155,9 +156,71 @@ def attach_eventuser(items):
     return file_name
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    Returns the suggested file name.
+    (Adapted from 'src/sentry/api/endpoints/organization_events.py')
```

Comment:
This is probably the important part. I've taken the logic that's in this file, and I'm repeating it in order to generate CSVs without a hard-limit of 50 or 100 rows.

 In terms of infrastructure, this task has its own queue with a concurrency of 1, so no worries there.

I'm just not sure if this code would work for _every_ type of discover query, since I'm not too knowledgeable on how complicated those queries can get. If it works, great! If i missed out on something big (like if you add `x` filter, with `y` range, let me know because I couldn't figure out too much)